### ISSUE TITLE
Timestamp Problems occured  when Upgraded to 2.1.2 

### ISSUE THREAD

**Issue by baturorkun** [fluent/fluent-bit#7339]
hello,

Some logs were stopped after upgrading to 2.1.2 I noticed that my "Trino" logs stream was stopped after upgrading and there was no error or warning. After opening stdout, I observed the difference.

My logs was like that before upgrading below.

[ 23423423.423424, { ....... } ]

Then after upgrading to 2.2.1 below.

[ [0.00000000 , {} ] ,  { ....... } ]

I could not understand why? 

thanks

**Comment by leonardo-albertovich** [comment]
Could you share your configuration file and some more context? Otherwise there isn't much we can say about it.

**Comment by baturorkun** [comment]
hello @leonardo-albertovich 

Meanwhile, I noticed that logs is coming but in different index after opened debug mode. Because timestamp  is Zero. When i downgrade 2.0.11, timestamp is right, not zero.

````
      [PARSER]
          Name        my_ui
          Format      regex
          Regex       (?<remote_addr>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}) - (?<remote_user>\S+) \[(?<time_local>[\w\/]+:\d{2}:\d{2}:\d{2} \+\d{4})\] "(?<request>\S+ \S+ \S+)" (?<status>\d{3}) (?<body_bytes_sent>\d+) "(?<http_referer>\S+)" "(?<http_user_agent>.+)" "(?<http_x_forwarded_for>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})" (?<request_time>\S+) (?<upstream_connect_time>\S+) (?<upstream_header_time>\S+) (?<upstream_response_time>\S+)$
          Time_Key    time
          Time_Format %Y-%m-%dT%H:%M:%S.%L%z
          Types       body_bytes_sent:integer request_time:float upstream_connect_time:float upstream_header_time:float upstream_response_time:float

      [PARSER]
          Name        multiline
          Format      regex
          Regex       ^(?<time>[^ ]+) (?<stream>stdout|stderr) (?<logtag>[^ ]*) (?<message>.*)$

      [PARSER]
          Name        trino_parser
          Format      regex
          Regex       (?<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z)\s+(?<log_level>\S+)\s+(?<thread>\S+)\s+(?<logger>\S+)\s+(?<message>.+)$

      [PARSER]
          Name        trino_query_parser
          Format      regex
          Regex       TIMELINE:\sQuery\s(?<query_id>\w+)\s::\s(?<query_result>[A-Z\s\_()]+)\s::\selapsed\s(?<query_elapse_time>\d+[a-z]+)\s::\splanning\s(?<query_planning_time>\d+[a-z]+)\s::\swaiting\s(?<query_waiting_time>\d+[a-z]+)\s::\sscheduling\s(?<query_scheduling_time>\d+[a-z]+)\s::\srunning\s(?<query_running_time>\d+[a-z]+)\s::\sfinishing\s(?<query_finishing_time>\d+[a-z]+)\s::\sbegin\s(?<query_begin_timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z)\s::\send\s(?<query_end_timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z)$

    inputs: |
      [INPUT]
          Name              tail
          Tag               kube.my_svc.*
          Path              /var/log/containers/**test-svc**.log
          Read_from_head    true
          Multiline         on
          Parser_Firstline  multiline
          DB                /var/log/flb_kube_my_svc.db
          Mem_Buf_Limit     10MB
          Skip_Long_Lines   Off
          Refresh_Interval  10
          Buffer_Chunk_Size 2MB
          Buffer_Max_Size   5MB

    outputs: |
      [OUTPUT]
          Name                es
          Match               kube.my_svc.*
          Host                elasticsearch
          Port                "9200"
          Logstash_Format     On
          Logstash_Prefix     my_svc
          Suppress_Type_Name  On
          Time_Key            @timestamp
          Replace_Dots        On
          Retry_Limit         False
          Buffer_Size         1024KB
          # HTTP_User           admin
          # HTTP_Passwd         admin
          tls                 Off
          tls.verify          Off

      [OUTPUT]
          Name    stdout
          Match   *

    filters: |
      # parse frontend services
      [FILTER]
          Name      parser
          Match     kube.my_svc.*
          Key_Name  message
          Parser    my_ui

      # parse backend services
      [FILTER]
          Name      parser
          Match     kube.my_svc.*
          Key_Name  message
          Parser    json

      # parse trino services
      [FILTER]
          Name      parser
          Match     kube.my_svc.*
          Key_Name  message
          Parser    trino_parser

      # parse trino query logs
      [FILTER]
          Name          parser
          Match         kube.my_svc.*
          Key_Name      message
          Parser        trino_query_parser
          Reserve_Data  True
          Preserve_Key  True

      # enrich with k8s metadata
      [FILTER]
          Name            kubernetes
          Match           *
          Buffer_Size     0
          Kube_Tag_Prefix kube.my_svc.var.log.containers.
          Labels          Off
          Annotations     Off

      [FILTER]
          Name          nest
          Match         kube.my_svc.*
          Operation     lift
          Nested_under  kubernetes
          Add_prefix    kubernetes.

      [FILTER]
          Name        record_modifier
          Match       kube.my_svc.*
          Remove_key  kubernetes.namespace_name
          Remove_key  kubernetes.docker_id
          Remove_key  kubernetes.container_hash
```

**Comment by lexxxel** [comment]
@leonardo-albertovich might be right. I have  the same issue and a downgrade to 2.0.x fixed the issue for me. Looks like we have to wait for a patch release :cry:

**Comment by leonardo-albertovich** [comment]
In these cases it's really important for the users to participate in the resolution of the issue and it makes a huge difference in the time it takes to get these fixed.

Would you be able to provide us with some context about your particular case @lexxxel? Would you be able to test the latest nightly build?

**Comment by lexxxel** [comment]
Of course, I can test the latest nightly build, if it is available as .deb package.
Context:
```conf
[INPUT]
    Name         forward
    Unix_Path    /run/fluent-bit.sock
    tag          log.mailserver
[FILTER]
    name          parser
    match         log.mailserver
    Parser        postfix_submission_smtpd
    Key_Name      log
    Reserve_Data  On
    Preserve_Key  On
[FILTER]
    name          parser
    match         log.mailserver
    Parser        postfix_smtpd
    Key_Name      log
    Reserve_Data  On
    Preserve_Key  On
[FILTER]
    name          parser
    match         log.mailserver
    Parser        postfix_smtp_send
    Key_Name      log
    Reserve_Data  On
    Preserve_Key  On
[FILTER]
    name          parser
    match         log.mailserver
    Parser        postfix_anvil
    Key_Name      log
    Reserve_Data  On
    Preserve_Key  On
[FILTER]
    name          parser
    match         log.mailserver
    Parser        postfix_postscreen
    Key_Name      log
    Reserve_Data  On
    Preserve_Key  On
[FILTER]
    name          parser
    match         log.mailserver
    Parser        dovecot
    Key_Name      log
    Reserve_Data  On
    Preserve_Key  On
[FILTER]
    name          parser
    match         log.mailserver
    Parser        postfix_qmgr_enqueue
    Key_Name      log
    Reserve_Data  On
    Preserve_Key  On
[OUTPUT]
    Name  opensearch
    Match log.mailserver
    Host  <IP>
    Port  9200
    Index log.mailserver
    HTTP_User <USER>
    HTTP_Passwd <PW>
    tls   On
    Suppress_Type_Name On
    Trace_Error On
    Buffer_Size 4M
```

parser
```
[PARSER]
          Name          postfix_submission_smtpd
          Format        regex
          Regex         ^(?:\w+\s+\d+\s+\S+\s\S+\s)(?<tool>postfix\/(?:submission|smtps)\/smtpd)(?:\[(?<pid>\d+)\])?:\ (?<smtpd>(?<state>connect|lost|disconnect|warning)?:?\ ?(?<message>[^\[\n]+)(?:\ ?\[(?<remote_addr>[^\]\n]+)\])?\ ?(?<detail>[^\n]+?)?)$
          types         pid:integer
          Time_Keep     On

      [PARSER]
          Name          postfix_smtpd
          Format        regex
          Regex         ^(?:\w+\s+\d+\s+\S+\s\S+\s)(?<tool>postfix\/local)(?:\[(?<pid>\d+)\]): (?<qid>[^:]+): to=<(?<to>[^>]+)>, relay=(?<relay>(none|[^\[]+))(\[(?<ip>[^\]]+)\]:(?<port>\d+))?, delay=(?<delay>[^,]+), delays=(?<receive>[^\/]+)\/(?<queue>[^\/]+)\/(?<conn>[^\/]+)\/(?<send>[^,]+), dsn=(?<dsn>[^,]+), status=(?<status>[^ ]+) \((?<msg>[^\)]+)\)
          Time_Keep     On
      [PARSER]
          name          postfix_smtp_send
          format        regex
          regex         ^(?:\w+\s+\d+\s+\S+\s\S+\s)(?<tool>postfix\/local)(?:\[(?<pid>\d+)\]): (?<qid>[^:]+): to=<(?<to>[^>]+)>, relay=(?<relay>(none|[^\[]+))(\[(?<ip>[^\]]+)\]:(?<port>\d+))?, delay=(?<delay>[^,]+), delays=(?<receive>[^\/]+)\/(?<queue>[^\/]+)\/(?<conn>[^\/]+)\/(?<send>[^,]+), dsn=(?<dsn>[^,]+), status=(?<status>[^ ]+) \((?<msg>[^\)]+)\)
          Time_Keep     On

      [PARSER]
          name          dovecot
          Format        regex
          regex         ^(?:\w+\s+\d+\s+\S+\s\S+\s)(?<tool>dovecot)?:\ (?<full>(?:(?<subtool>[^\(\:]+)?(?:\((?<subtool_user>[^\)\:]+)\)(?:<\S+><\S+>)?)?):\ ?(?<message>(?:Login:\ user=<(?<user>[^\>]+)>,\ method=(?<method>[^\,]+),\ rip=(?<remote_addr>[^\,]+),\ lip=(?<local_addr>[^\,]+),\ mpid=(?<pid>\d+),\ (?<security>[^\,]+), session=(?<session>[^\,\n]+))|(sieve: (?<sieve>msgid=<(?<msgid>[^\>]+)>.*?'(?<mailbox>[^\']+)'))|[^\[\n]+))$
          Time_Keep     On

      [PARSER]
          name          postfix_anvil
          Format        regex
          regex         ^(?:\w+\s+\d+\s+\S+\s\S+\s)(?<tool>postfix\/anvil)\[(?<pid>\d+)\]:\ (?<full>statistics:\ ((?:max connection (?:rate (?<connection_rate>\S+))?(?:count (?<connection_count>\d+))? for \(submission:(?<remote_addr>[^\)]+)\))|(?:max cache size (?<cache_size>\d+)))(?<message>.*))$
          Time_Keep     On

      [PARSER]
          name          postfix_postscreen
          Format        regex
          regex         ^(?:\w+\s+\d+\s+\S+\s\S+\s)(?<tool>postfix\/postscreen)(?:\[(?<pid>\d+)\])?:\ (?<postscreen>(?<state>\S+) (?<message>[^\[\n]*?(?:\[(?<remote_addr>[^\]]+)\]:(?<remote_port>\d+))(?: to \[(?<local_addr>[^\]]+)\]:(?<local_port>\d+))?[^\[\n]*?))$
          types         remote_port:integer local_port:integer
          Time_Keep     On

      [PARSER]
          name          postfix_qmgr_enqueue
          format        regex
          regex         ^(?:\w+\s+\d+\s+\S+\s\S+\s)(?<tool>postfix\/qmgr)(?:\[(?<pid>\d+)\]): (?<qid>[^:]+): from=<(?<from>[^>]*)>, size=(?<size>\d+), nrcpt=(?<nrcpt>\d+) \((?<queue>[^\)]+)\)
          types         size:integer nrcpt:integer
          Time_Keep     On
```
Addition: the `Time_keep` has been added in attend to fix the issue. It does not make a difference.

fluentbit 2.1.2
```
May 09 13:38:35 mailserver-xxxxxxxx fluent-bit[49230]: [1] log.mailserver: [[0.000000000, {}], {"tool"=>"dovecot", "full"=>"imap(xxxxxxxxxx)<15673><uZpr3kL7eJYKAAAC>: Disconnected: Connection closed (SEARCH finished 0.083 secs ago) in=3441 out=58733 deleted=0 expunged=0 trashed=0 hdr_count=2 hdr_bytes=705 body_count=0 body_bytes=0", "subtool"=>"imap", "subtool_user"=>"xxxxxxxxxxxxxxxx, "message"=>"Disconnected: Connection closed (SEARCH finished 0.083 secs ago) in=3441 out=58733 deleted=0 expunged=0 trashed=0 hdr_count=2 hdr_bytes=705 body_count=0 body_bytes=0", "container_id"=>"9dab16ff2bdf1bf9327068c693944d105cb83b5d1cc45952d09a65c9d49865d1", "container_name"=>"/mailserver", "source"=>"stdout", "log"=>"May  9 15:38:23 mail dovecot: imap(xxxxxxxxxxx)<15673><uZpr3kL7eJYKAAAC>: Disconnected: Connection closed (SEARCH finished 0.083 secs ago) in=3441 out=58733 deleted=0 expunged=0 trashed=0 hdr_count=2 hdr_bytes=705 body_count=0 body_bytes=0", "host"=>"xxxxxxxxxxxxxxx"}]
```

fluentbit 2.0.11
```
May 09 13:37:11 mailserver-xxxxxxxxx fluent-bit[48218]: [1] log.mailserver: [1683639424.000000000, {"tool"=>"postfix/postscreen", "pid"=>"15225", "postscreen"=>"PREGREET 11 after 0.01 from [IP]:53390: EHLO User\r\n", "state"=>"PREGREET", "message"=>"11 after 0.01 from [IP]:53390: EHLO User\r\n", "remote_addr"=>"IP", "remote_port"=>53390, "container_id"=>"9dab16ff2bdf1bf9327068c693944d105cb83b5d1cc45952d09a65c9d49865d1", "container_name"=>"/mailserver", "source"=>"stdout", "log"=>"May  9 15:37:04 mail postfix/postscreen[15225]: PREGREET 11 after 0.01 from [IP]:53390: EHLO User\r\n", "host"=>"xxxxxxxxxxxxxxxxx"}]
```

**Comment by lexxxel** [comment]
Don't mind:

```
May 09 14:05:38 mailserver-xxxx fluent-bit[63126]: Fluent Bit v2.1.3
May 09 14:05:38 mailserver-xxxx fluent-bit[63126]: * Copyright (C) 2015-2022 The Fluent Bit Authors
May 09 14:05:38 mailserver-xxxx fluent-bit[63126]: * Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
May 09 14:05:38 mailserver-xxxx fluent-bit[63126]: * https://fluentbit.io
May 09 14:05:53 mailserver-xxxx fluent-bit[63126]: [0] log.mailserver: [[1683641139.000000000, {}], {"tool"=>"dovecot", "full"=>"imap
```

**Comment by leonardo-albertovich** [comment]
Well, that's interesting because unless I'm missing some context you are not using the json parser which means it shouldn't be the same bug.

I'll try to check the change log as time permits but it would be really helpful if you could keep testing if a bit more so we know if your truly issue is fixed or not.

**Comment by baturorkun** [comment]
We needed a feature from 2.1.2 so we did not want to downgrade, so for a workaround, set Current_Time_Index = On In this way logs can stream to the right index file. We are storing them as daily log files. The log files streamed to "index_0.0.0" because of this bug. But Current_Time_Index solved index name problems.

**Comment by leonardo-albertovich** [comment]
@lexxxel I saw yours and acknowledged it but I wasn't sure if the original issue was caused by the same bug which is why I wanted to be sure it's fixed before closing the issue. Better safe than sorry!

Thanks a lot for staying on top of it though, user collaboration makes the process **a lot** simpler and less painful.

**Comment by nwnt** [comment]
I'm observing the same issue as well using the dummy input:
```
docker run -it fluent/fluent-bit /fluent-bit/bin/fluent-bit -i dummy -o stdout
Fluent Bit v2.1.2
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2023/05/12 05:32:51] [ info] [fluent bit] version=2.1.2, commit=113c7c00d2, pid=1
[2023/05/12 05:32:51] [ info] [storage] ver=1.4.0, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2023/05/12 05:32:51] [ info] [cmetrics] version=0.6.1
[2023/05/12 05:32:51] [ info] [ctraces ] version=0.3.0
[2023/05/12 05:32:51] [ info] [input:dummy:dummy.0] initializing
[2023/05/12 05:32:51] [ info] [input:dummy:dummy.0] storage_strategy='memory' (memory only)
[2023/05/12 05:32:51] [ info] [sp] stream processor started
[2023/05/12 05:32:51] [ info] [output:stdout:stdout.0] worker #0 started
[0] dummy.0: [[1683869571.733871004, {}], {"message"=>"dummy"}]
[0] dummy.0: [[1683869572.733746205, {}], {"message"=>"dummy"}]
[0] dummy.0: [[1683869573.733607250, {}], {"message"=>"dummy"}]
[0] dummy.0: [[1683869574.733450531, {}], {"message"=>"dummy"}]
[0] dummy.0: [[1683869575.733628131, {}], {"message"=>"dummy"}]
```

**Comment by leonardo-albertovich** [comment]
Could you please elaborate? I don't see what's abnormal in your output @nwnt.

**Comment by nwnt** [comment]
Hi @leonardo-albertovich, sorry I was out.  After 2.1, the timestamp field is in an array instead of value (italic):

[0] dummy.0: _[[1683869575.733628131, {}]_, {"message"=>"dummy"}]

Here is the same output from the exact same command in version 2.0.11

```
Fluent Bit v2.0.11
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2023/05/13 18:17:28] [ info] [fluent bit] version=2.0.11, commit=bf22d53a69, pid=1
[2023/05/13 18:17:28] [ info] [storage] ver=1.4.0, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2023/05/13 18:17:28] [ info] [cmetrics] version=0.6.0
[2023/05/13 18:17:28] [ info] [ctraces ] version=0.3.0
[2023/05/13 18:17:28] [ info] [input:dummy:dummy.0] initializing
[2023/05/13 18:17:28] [ info] [input:dummy:dummy.0] storage_strategy='memory' (memory only)
[2023/05/13 18:17:28] [ info] [sp] stream processor started
[2023/05/13 18:17:28] [ info] [output:stdout:stdout.0] worker #0 started
[0] dummy.0: [1684001848.733694266, {"message"=>"dummy"}]
[0] dummy.0: [1684001849.733587692, {"message"=>"dummy"}]
[0] dummy.0: [1684001850.733591871, {"message"=>"dummy"}]
[0] dummy.0: [1684001851.733666742, {"message"=>"dummy"}]
[0] dummy.0: [1684001852.733491556, {"message"=>"dummy"}]
[0] dummy.0: [1684001853.733646276, {"message"=>"dummy"}]
```

**Comment by leonardo-albertovich** [comment]
Yes, that was a deliberate change, starting with 2.1 the first member is a 2 element array where the first one is the timestamp and the second one is the metadata map.

**Comment by edespong** [comment]
@leonardo-albertovich I am experiencing what I think is the same issue and would like to verify on nightly, but cannot find it. Could you point me in the right direction?

**Comment by leonardo-albertovich** [comment]
Could you elaborate @edespong?

**Comment by edespong** [comment]
> Could you elaborate @edespong?

You wrote above: 
```
In these cases it's really important for the users to participate in the resolution of the issue and it makes a huge difference in the time it takes to get these fixed.

Would you be able to provide us with some context about your particular case @lexxxel? Would you be able to test the latest nightly build?
```

I would also like to test that my, very similar, issues are solved on the nightly build. I cannot, however, figure out where to find the built executable for the nightly build.

**Comment by leonardo-albertovich** [comment]
The issue here is that this is a rather old conversation and what was included in the nightly build at the moment was released in fluent-bit 2.1.4 (I can't remember if 2.1.3 had it as well) so if you are having similar issues now it would be better if you opened a new issue with all of the appropriate context information we need to properly triage your case and address any issues we identify.

**Comment by edespong** [comment]
Thanks. I validated the fix on 2.1.4. I think I was thrown off by the fact that this issue was still Open and in waiting-for-triage status. Sorry for the confusion.

**Comment by leonardo-albertovich** [comment]
Thanks for confirming it, I'll close the issue now to avoid confusing people in the future.
